### PR TITLE
fix: [CI-9771]: Taking Pod as Base64 object for logging purpose

### PR DESCRIPTION
**Pull request by ShobhitSingh11**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53975)

### CODE REVIEW

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 1]
```diff
@@ -626,4 +630,19 @@ private void updateContainerWithEnvVariable(Set<String> allSecrets, ContainerPar
...
+      ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+      ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteArrayOutputStream);
+
+      objectOutputStream.writeObject(object);
+      objectOutputStream.close();
+
+      byte[] objectBytes = byteArrayOutputStream.toByteArray();
+      return Base64.getEncoder().encodeToString(objectBytes);
+    } catch (Exception e) {
+      e.printStackTrace();
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:644]: put it inside 
log.error()

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Review by harshjain12 — DISMISSED** [review]

**Comment by harshjain12** [comment]
what is the size of base64 string ?

**Comment by harshjain12** [comment]
can you attach screenshot of log from your local

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
> what is the size of base64 string ?

Ref : https://stackoverflow.com/questions/13378815/base64-length-calculation

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger ut4, trigger ut8

**Comment by ShobhitSingh11** [comment]
trigger ut8

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseStep.java [hunk 2]
```diff
@@ -84,7 +86,11 @@ private String getChildNodeId(Ambiance ambiance, CodeBaseStepParameters stepPara
   @Override
   public StepResponse handleChildResponse(
       Ambiance ambiance, CodeBaseStepParameters stepParameters, Map<String, ResponseData> responseDataMap) {
-    log.info("Completed execution for codebase node step [{}]", stepParameters);
+    try {
+      log.info("Completed execution for codebase node step [{}]", EncodingUtils.convertToBase64String(stepParameters));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseStep.java:92]: Do not throw exception

Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseStep.java:92]: log.error()

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/V1/InitializeTaskStepV2.java [hunk 3]
```diff
@@ -312,7 +314,11 @@ public String executeBuild(Ambiance ambiance, StepElementParameters stepParamete
 
     // Secrets are in decrypted format for DLITE_VM type
     if (buildSetupTaskParams.getType() != DLITE_VM) {
-      log.info("Created params for build task: {}", buildSetupTaskParams);
+      try {
+        log.info("Created params for build task: {}", EncodingUtils.convertToBase64String(buildSetupTaskParams));
+      } catch (IOException e) {
+        throw new RuntimeException(e);
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/V1/InitializeTaskStepV2.java:320]: do not throw exception

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/SecurityStageStepPMS.java [hunk 4]
```diff
@@ -68,8 +70,12 @@ public ChildExecutableResponse obtainChild(
     String accountId = AmbianceUtils.getAccountId(ambiance);
     String projectIdentifier = AmbianceUtils.getProjectIdentifier(ambiance);
 
-    log.info("Executing integration stage with params accountId {} projectId {} [{}]", accountId, projectIdentifier,
-        stepParameters);
+    try {
+      log.info("Executing integration stage with params accountId {} projectId {} [{}]", accountId, projectIdentifier,
+          EncodingUtils.convertToBase64String(stepParameters));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/SecurityStageStepPMS.java:77]: do not throw exeption

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/STOSpecStep.java [hunk 5]
```diff
@@ -32,14 +34,22 @@ public class STOSpecStep implements ChildExecutable<IntegrationStageStepParamete
   @Override
   public ChildExecutableResponse obtainChild(
       Ambiance ambiance, IntegrationStageStepParametersPMS stepParameters, StepInputPackage inputPackage) {
-    log.info("Starting execution for spec node step [{}]", stepParameters);
+    try {
+      log.info("Starting execution for spec node step [{}]", EncodingUtils.convertToBase64String(stepParameters));
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/STOSpecStep.java:38]: do not change it

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/STOSpecStep.java [hunk 6]
```diff
@@ -32,14 +34,22 @@ public class STOSpecStep implements ChildExecutable<IntegrationStageStepParamete
...
+    }
     return ChildExecutableResponse.newBuilder().setChildNodeId(stepParameters.getChildNodeID()).build();
   }
 
   @Override
   public StepResponse handleChildResponse(
       Ambiance ambiance, IntegrationStageStepParametersPMS stepParameters, Map<String, ResponseData> responseDataMap) {
-    log.info("Completed execution for spec node step [{}]", stepParameters);
+    try {
+      log.info("Completed execution for spec node step [{}]", EncodingUtils.convertToBase64String(stepParameters));
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/STOSpecStep.java:49]: do not change it

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/SecurityStageStepPMS.java [hunk 7]
```diff
@@ -68,8 +70,12 @@ public ChildExecutableResponse obtainChild(
     String accountId = AmbianceUtils.getAccountId(ambiance);
     String projectIdentifier = AmbianceUtils.getProjectIdentifier(ambiance);
 
-    log.info("Executing integration stage with params accountId {} projectId {} [{}]", accountId, projectIdentifier,
-        stepParameters);
+    try {
+      log.info("Executing integration stage with params accountId {} projectId {} [{}]", accountId, projectIdentifier,
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/SecurityStageStepPMS.java:74]: do not change it

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/SecurityStageStepPMS.java (hunk 4 above)
Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/SecurityStageStepPMS.java:77]: do not change it

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseStep.java (hunk 2 above)
Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/codebase/CodeBaseStep.java:92]: do not change it

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

File: 980-commons/src/main/java/io/harness/serializer/JsonUtils.java [hunk 8]
```diff
@@ -106,6 +106,7 @@ public Set<Option> options() {
     mapper.registerModule(new Jdk8Module());
     mapper.registerModule(new GuavaModule());
     mapper.registerModule(new JavaTimeModule());
+    mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
```
Comment (harshjain12) [inline 980-commons/src/main/java/io/harness/serializer/JsonUtils.java:109]: Do not change it

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 9]
```diff
@@ -174,8 +174,7 @@ public K8sTaskExecutionResponse executeTaskInternal(
         log.info("Setting up pod spec");
         V1Pod pod = podSpecBuilder.createSpec(podParams).build();
         updatePodWithDelegateVolumes(coreV1Api, namespace, pod);
-
-        log.info("Creating pod with spec: {}", pod);
+        log.info("Creating pod with spec: {}", EncodingUtils.convertToBase64String(pod));
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:177]: put it in try and catch

File: 980-commons/src/main/java/io/harness/data/encoding/BUILD.bazel [hunk 10]
```diff
@@ -20,6 +20,7 @@ java_library(
         "//:lombok",
         "//980-commons/src/main/java/io/harness/data/structure:module",
         "//980-commons/src/main/java/io/harness/exception:module",
+        "//980-commons/src/main/java/io/harness/serializer:module",
```
Comment (harshjain12) [inline 980-commons/src/main/java/io/harness/data/encoding/BUILD.bazel:23]: do we need it ?

Comment (ShobhitSingh11) [reply 980-commons/src/main/java/io/harness/data/encoding/BUILD.bazel:23]: for JsonUtils build

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger ut9

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger messagecheck

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java [hunk 11]
```diff
@@ -109,8 +111,21 @@ public Class<StageElementParameters> getStepParametersClass() {
   public ChildExecutableResponse obtainChild(
       Ambiance ambiance, StageElementParameters stepParameters, StepInputPackage inputPackage) {
     NGAccess ngAccess = AmbianceUtils.getNgAccess(ambiance);
-    log.info("Executing integration stage with params accountId {} projectId {} [{}]", ngAccess.getAccountIdentifier(),
-        ngAccess.getProjectIdentifier(), stepParameters);
+
+    // Here have to write it separately because we can't import RecastOrchestrationUtils in Encoding utils class.
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java:115]: remove comments

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/InitializeTaskStep.java [hunk 12]
```diff
@@ -264,7 +266,11 @@ public TaskRequest obtainTaskAfterRbac(
 
     // Secrets are in decrypted format for DLITE_VM type
     if (buildSetupTaskParams.getType() != DLITE_VM) {
-      log.info("Created params for build task: {}", buildSetupTaskParams);
+      try {
+        log.info("Created params for build task: {}", EncodingUtils.convertToBase64String(buildSetupTaskParams));
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/InitializeTaskStep.java:270]: Not required

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java [hunk 13]
```diff
@@ -109,8 +111,21 @@ public Class<StageElementParameters> getStepParametersClass() {
...
+    // so can't add 953-yaml-commons inside 980-commons build file as well because it will create an infinite cycle.
+    // RecastOrchestrationUtils lies inside 953-yaml-commons and Encoding utils lies inside 980-commons.
+    try {
+      String jsonString = RecastOrchestrationUtils.toJson(stepParameters);
+      byte[] jsonByte = EncodingUtils.compressString(jsonString);
+      String base64String = EncodingUtils.encodeBase64(jsonByte);
+      log.info("Executing integration stage with params accountId {} projectId {} [{}]",
+          ngAccess.getAccountIdentifier(), ngAccess.getProjectIdentifier(), base64String);
+    } catch (IOException e) {
+      log.error("Could not serialize", e);
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java:127]: Could not serialize Class name

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/V1/InitializeTaskStepV2.java [hunk 14]
```diff
@@ -312,7 +314,11 @@ public String executeBuild(Ambiance ambiance, StepElementParameters stepParamete
 
     // Secrets are in decrypted format for DLITE_VM type
     if (buildSetupTaskParams.getType() != DLITE_VM) {
-      log.info("Created params for build task: {}", buildSetupTaskParams);
+      try {
+        log.info("Created params for build task: {}", EncodingUtils.convertToBase64String(buildSetupTaskParams));
+      } catch (IOException e) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/V1/InitializeTaskStepV2.java:319]: handle exception

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java (hunk 13 above)
Comment (harshjain12) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/states/IntegrationStageStepPMS.java:127]: handle all exception

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java [hunk 15]
```diff
@@ -175,7 +175,12 @@ public K8sTaskExecutionResponse executeTaskInternal(
         V1Pod pod = podSpecBuilder.createSpec(podParams).build();
         updatePodWithDelegateVolumes(coreV1Api, namespace, pod);
 
-        log.info("Creating pod with spec: {}", pod);
+        try {
+          log.info("Creating pod with spec: {}", EncodingUtils.convertToBase64String(pod));
+        } catch (IOException e) {
```
Comment (harshjain12) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/citasks/cik8handler/CIK8InitializeTaskHandler.java:180]: handle all exception

**Review by harshjain12 — CHANGES_REQUESTED** [review]

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Comment by ShobhitSingh11** [comment]
trigger smartchecks

**Comment by ShobhitSingh11** [comment]
trigger uts

**Review by harshjain12 — APPROVED** [review]

**Pull request merged**